### PR TITLE
[actions] Run gazelle and go mod tidy on PRs

### PR DESCRIPTION
**Pull request by JamesMBartlett**
Summary: Run `make gazelle` and `make go-setup` on PRs and fail if there are any changes as a result.

Type of change: /kind test-infra

Test Plan: Testing with this PR to see if a change to `go.mod` causes a check failure.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module px.dev/pixie
 go 1.20
 
 require (
-	cloud.google.com/go v0.81.0
+	cloud.google.com/go v0.81.1
```
Comment (vihangm) [inline go.mod:6]: seems like an unrelated change?

**Review by vihangm — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (JamesMBartlett) [reply go.mod:6]: I'm testing it
